### PR TITLE
Text Example: ThisItem

### PR DESCRIPTION
**Pull request by MikeStall**
Add a test showing how to support ThisItem  in a loop using the SymbolTable classes.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs [hunk 1]
```diff
@@ -618,6 +618,56 @@ public async Task TestMismatch2()
...
 
+        // Demonstrate how to use ThisItem in a loop.
+        // All Loop iterations can access common global state,
+        // but also have their own loop-specific identifer. 
+
+        [Fact]
+        public async Task ThisItemTest()
+        {
+            var symTableGlobals = new SymbolTable { DebugName = "Globals" };
+            var slotGlobal = symTableGlobals.AddVariable("globalvar", FormulaType.Number);
```
Comment (jas-valgotar) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs:629]: >globalvar [](http://example.com/codeflow?start=58&length=9)

super nit: rename to `globalVar` for readability?

**Review by jas-valgotar — APPROVED** [review]

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs [hunk 2]
```diff
@@ -618,6 +618,56 @@ public async Task TestMismatch2()
...
+
+            // Loop body can be run in parallel since each eval is independent. 
+            // An there are no mutations on shared state.
+            Parallel.For(0, 10, (i) =>
+            {
+                // Each loop body gets its own iteration-specific values.
+                var symValuesThisItem = symTableThisItem.CreateValues();
+                symValuesThisItem.Set(slotThisItem, FormulaValue.New(i));
+
+                var symValues = symTable.CreateValues(symValuesGlobals, symValuesThisItem);
```
Comment (anderson-joyle) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs:661]: >symValuesGlobals [](http://example.com/codeflow?start=54&length=16)

Is symValuesGlobals immutable? If it is, would be possible to add a new test case showing what would it happen if something tries to change it in a lopp?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by anderson-joyle — APPROVED** [review]
:shipit:

**Pull request merged**